addons/pos_restaurant/models/pos_restaurant.py

```diff
@@ -94,35 +94,43 @@ class RestaurantTable(models.Model):
     color = fields.Char('Color', help="The table's color, expressed as a valid 'background' CSS property value")
     active = fields.Boolean('Active', default=True, help='If false, the table is deactivated and will not be available in the point of sale')
 
+    # multi talbe compute
     def are_orders_still_in_draft(self):
-        draft_orders = self.env['pos.order'].search([('table_id', '=', self.id), ('state', '=', 'draft')])
+        draft_orders = self.env['pos.order'].search([('table_id', 'in', self.ids), ('state', '=', 'draft')])
         return len(draft_orders) > 0
 
-    @api.model
-    def create_from_ui(self, table):
-        """ create or modify a table from the point of sale UI.
-            table contains the table's fields. If it contains an
-            id, it will modify the existing table. It then
-            returns the id of the table.
-        """
-        if table.get('id'):
-            table_id = table['id']
-
-        if table.get('floor_id'):
-            table['floor_id'] = table['floor_id'][0]
-
-        sanitized_table = dict([(key, val) for key, val in table.items() if key in self._fields and val is not None])
-        table_id = sanitized_table.pop('id', False)
-        if table_id:
-            self.browse(table_id).write(sanitized_table)
-        else:
-            table_id = self.create(sanitized_table).id
-
-        sanitized_table['id'] = table_id
-        for config_id in self.env['pos.config'].search([('floor_ids.table_ids', 'in', table_id)]):
-            self.env['bus.bus']._sendone(f'pos_config-{config_id.id}', 'table_changed', {'changes': sanitized_table})
-
-        return table_id
+    def multi_write(self, table_list):
```

Comment:
rename `table_list` to `table_vals`. And it would be nice if you can type-annotate it. Because normally, `_vals` are list of objects. In your case, it's a mapping (dict) of table id (in string) to the corresponding object.